### PR TITLE
Verify suite - default behaviour

### PR DESCRIPTION
**Pull request by zajck**
This PR introduces some changes in verify suite:

- contract list to be verified is supplied in file `scripts/config/contract-verification.js` and not hardcoded in hardhat config
- if contract list is empty it tries to verify everything
- just slightly changed what env represents so it's in line with other scripts
- task.md was updated to mention how to verify only a subset of contracts

### CODE REVIEW

**Review by hswopeams — CHANGES_REQUESTED** [review]

File: scripts/config/contract-verification.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * List of contract to verify on explorer, using verify-suite script
```
Comment (cliffhall) [inline scripts/config/contract-verification.js:2]: ~~contract~~  contracts

File: scripts/config/contract-verification.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * Process:
+ *  1.  Edit scripts/config/contract-verification.js. Addresses will be pulled from /addresses/<chainId>-<network>.json or environments file
+ *  1a. If you want to verify all contract, leave contractList empty
```
Comment (cliffhall) [inline scripts/config/contract-verification.js:6]: ~~contract~~  contracts

File: scripts/config/contract-verification.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+ *  1b. If you want to verify only a subset of contract, specify them in contractList.
```
Comment (cliffhall) [inline scripts/config/contract-verification.js:7]: ~~contract~~  contracts

File: scripts/config/contract-verification.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ *      Use names of actual implementations, not interfaces.
+ *  2. Run the appropriate npm script in package.json to verify contract for a given network and environment
```
Comment (cliffhall) [inline scripts/config/contract-verification.js:9]: ~~contract~~  contracts

File: scripts/util/verify-suite.js [hunk 5]
```diff
@@ -1,34 +1,39 @@
...
+ *  e.g.: npx hardhat verify-suite --network polygon --chain-id 137 --env prod
+ *      : reads addresses/137-polygon-prod.json
  *      : verifies each contract listed, with the given constructor args
  *
- *  If you need to filter the list of contracts to verify,
- *  edit hardhat.config.js temporarily and set the filter
- *  array in the task definition
+ * Process:
+ *  1.  Edit scripts/config/contract-verification.js. Addresses will be pulled from /addresses/<chainId>-<network>.json or environments file
+ *  1a. If you want to verify all contract, leave contractList empty
```
Comment (cliffhall) [inline scripts/util/verify-suite.js:17]: ~~contract~~  contracts

File: scripts/util/verify-suite.js [hunk 6]
```diff
@@ -1,34 +1,39 @@
...
+ *  1b. If you want to verify only a subset of contract, specify them in contractList.
```
Comment (cliffhall) [inline scripts/util/verify-suite.js:18]: ~~contract~~  contracts

File: scripts/util/verify-suite.js [hunk 7]
```diff
@@ -1,34 +1,39 @@
...
+ *      Use names of actual implementations, not interfaces.
+ *  2. Run the appropriate npm script in package.json to verify contract for a given network and environment
```
Comment (cliffhall) [inline scripts/util/verify-suite.js:20]: ~~contract~~  contracts

**Review by cliffhall — CHANGES_REQUESTED** [review]
Just a few typos

File: scripts/util/verify-suite.js [hunk 8]
```diff
@@ -1,34 +1,39 @@
...
- *      : reads addresses/137-polygon.json
+ *  e.g.: npx hardhat verify-suite --network polygon --chain-id 137 --env prod
+ *      : reads addresses/137-polygon-prod.json
  *      : verifies each contract listed, with the given constructor args
  *
- *  If you need to filter the list of contracts to verify,
- *  edit hardhat.config.js temporarily and set the filter
- *  array in the task definition
+ * Process:
+ *  1.  Edit scripts/config/contract-verification.js. Addresses will be pulled from /addresses/<chainId>-<network>.json or environments file
```
[the remark this answers is not in the document]
Comment (zajck) [reply scripts/util/verify-suite.js]: fixed

File: scripts/util/verify-suite.js (hunk 5 above)
Comment (zajck) [reply scripts/util/verify-suite.js]: fixed

File: scripts/util/verify-suite.js (hunk 6 above)
Comment (zajck) [reply scripts/util/verify-suite.js]: fixed

File: scripts/util/verify-suite.js (hunk 7 above)
Comment (zajck) [reply scripts/util/verify-suite.js]: fixed

File: scripts/util/verify-suite.js [hunk 9]
```diff
@@ -1,34 +1,39 @@
+const hre = require("hardhat");
```
[the remark this answers is not in the document]
Comment (zajck) [reply scripts/util/verify-suite.js]: Makes sense to move it to the top level, since it's role is closer to those scripts than utils.

Moved.

**Review by hswopeams — APPROVED** [review]

**Review by anajuliabit — APPROVED** [review]

**Pull request merged**